test/data/testSP.py

```diff
@@ -583,6 +584,28 @@ class OrcaSPTest(GenericSPTest):
     num_scf_criteria = 3
 
 
+class NBOSPTest(GenericSPTest):
+    """Customized restricted single point unittest"""
+    def testpopulations(self):
+        assert self.data.populations.keys == list(['nao', 'atom', 'no', 'lang', 'type', 'occupancy', 'energy']) 
+        assert isinstance(self.data.populations['nao'], List[int])

```

Comment:
I'm not sure this test is getting run, because
```
>>> l = [1, 2, 3]
>>> import typing
>>> isinstance(l, typing.List)
True
>>> isinstance(l, typing.List[int])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ejberqu/.pyenv/versions/3.10.11/lib/python3.10/typing.py", line 994, in __instancecheck__
    return self.__subclasscheck__(type(obj))
  File "/Users/ejberqu/.pyenv/versions/3.10.11/lib/python3.10/typing.py", line 997, in __subclasscheck__
    raise TypeError("Subscripted generics cannot be used with"
TypeError: Subscripted generics cannot be used with class and instance checks
```
`typing` is also not for runtime type checking, if you wanted to do that over abstract container types it would be from `collections.abc`, and then checking element types would be a list comprehension over the container.